Docs

m-reuter opened the pull request:
This PR is for us to discuss open issues in the doc branch.

agirodi committed:
docs add/edit singularity-docs

Change and improve documentation on singularity usage using local and remote images.
Also some readability changes

agirodi committed:
docs edit fs-license

Add that --fs-license is not needed for the CNN

m-reuter committed:
remove req. abs. path

m-reuter committed:
copy over formatting

m-reuter committed:
formatting

m-reuter committed:
formatting and some adjustment

m-reuter committed:
use t1-weighted.nii.gz as input examples

m-reuter committed:
for local example remove deepni

m-reuter committed:
fix cpu and remove hard coded version

m-reuter committed:
formatting and minor details added

m-reuter committed:
update overview

m-reuter committed:
re-order, update self-build names etc.

m-reuter committed:
update info

m-reuter committed:
update to new directoy structure

m-reuter committed:
updates

m-reuter committed:
cnn -> vinn

m-reuter commented on FastSurferCNN/README.md:
```diff
@@ -10,66 +10,77 @@ The network was trained with conformed images (UCHAR, 1-0.7 mm voxels and standa
-* --ckpt_sag: path to sagittal network checkpoint
-* --ckpt_cor: path to coronal network checkpoint
-* --ckpt_ax: path to axial network checkpoint
+* `--ckpt_sag`: path to sagittal network checkpoint
```
can (or should) ckpt even be passed any longer??

dkuegler replied:
The run_prediction.py script still accepts paths for checkpoints and configs. Just run_fastsurfer.py does not any more.

m-reuter committed:
surfreg is now on by default

m-reuter committed:
remove no_aprarc and surfreg on by default

dkuegler commented on FastSurferCNN/README.md:
```diff
@@ -10,66 +10,77 @@ The network was trained with conformed images (UCHAR, 1-0.7 mm voxels and standa
 --t subjectX \
---seg_log ../data/temp_Competitive.log \
+--seg_log ../output/temp_Competitive.log \
 --ckpt_sag ../checkpoints/aparc_vinn_sagittal_v2.0.0.pkl \
```
But the example should probably not have the checkpoints any more. It should just use the defaults.

agirodi replied:
This is resolved. See [here](https://github.com/Deep-MI/FastSurfer/pull/231)

dkuegler commented on FastSurferCNN/README.md:
```diff
@@ -10,66 +10,77 @@ The network was trained with conformed images (UCHAR, 1-0.7 mm voxels and standa
---seg_log ../data/temp_Competitive.log \
+--sd ../output \
+--seg_log ../output/temp_Competitive.log \
 --ckpt_sag ../checkpoints/aparc_vinn_sagittal_v2.0.0.pkl \
```
Same here

agirodi replied:
This is resolved. See [here](https://github.com/Deep-MI/FastSurfer/pull/231)

dkuegler commented on README.md:
```diff
@@ -33,56 +33,44 @@ List them by running the following command:
+* `--viewagg_device`: Define where the view aggregation should be run on. Can be "auto" or a device (see --device). By default, the program checks if you have enough memory to run the view aggregation on the gpu. The total memory is considered for this decision. If this fails, or you actively overwrote the check with setting with "cpu" view agg is run on the cpu. Equivalently, if you pass a diffe ...
+* `--device`: Select device for NN segmentation ("cpu", "cuda"), where cuda means Nvidia GPU, you can select which one "cuda:1". Default: "auto", check GPU and then CPU
+* `--batch`: Batch size for inference. Default: 1. 
+* `--vol_segstats`: Additionally return volume-based aparc.DKTatlas+aseg statistics for DL-based segmentation (does not  require surfaces). Can be used in combination with `--seg_only` in which case recon-surf only runs till CC is added (akin to --seg_with_cc_only).
```
`--seg_with_cc_only` does not sense any more. This option is removed.
Maybe this is also wrong in run_fastsurfer.sh

agirodi replied:
This is resolved. See [here](https://github.com/Deep-MI/FastSurfer/pull/231)

dkuegler commented on README.md:
```diff
@@ -33,56 +33,44 @@ List them by running the following command:
-* --config_cor: Path to the axial coronal file
-* --ignore_fs_version: Switch on to avoid check for FreeSurfer version. Program will terminate if the supported version (see recon-surf.sh) is not sourced. Can be used for testing dev versions.
-* --help: Prints helptext out
+* `--vox_size 1|auto`:  Force processing at a specific voxel size. Only "auto" and "1" are supported values. If "1" is specified, the T1w image will be conformed to 1mm voxel size and processed, if "auto" is specified (default), the T1w image will be conformed isometric voxels of the smallest voxel edge in the image for highres processing.
```
the vox_size option was changed. run_fastsurfer.sh has updated docs

agirodi replied:
This is resolved. See [here](https://github.com/Deep-MI/FastSurfer/pull/231)

dkuegler commented on recon_surf/README.md:
```diff
@@ -4,11 +4,10 @@ This directory contains all information needed to run the surface-generation and
 alternative for cortical surface reconstruction, mapping of cortical labels and traditional point-wise and ROI thickness analysis.
 The basis for the reconstruction pipeline is the accurate anatomical whole brain segmentation following the DKTatlas 
-such as the one provided by the FastSurferCNN deep learning architecture.
+such as the one provided by the FastSurferCNN or FastSurferVINN deep learning architectures.
```
This documentation probably needs some details on the behavior w.r.t. hires processing.
Currently, there is no dedicated flag, but all segmentation files that are passed, which have a voxel size smaller than 0.999 are treated as hires.

dkuegler commented on run_fastsurfer.sh:
```diff
@@ -115,16 +114,16 @@ FLAGS:
                             stored, but only the merged segmentation will be stored
                             (see --main_segfile <filename>).
-                            Requires an ABSOLUTE Path! Default location:
+                            Default location:
```
Has this changed? does this now "not" require an absolute path?

dkuegler commented on FastSurferCNN/README.md:
```diff
@@ -10,66 +10,77 @@ The network was trained with conformed images (UCHAR, 1-0.7 mm voxels and standa
+                    If this fails, or you actively overwrote the check with setting `--viewagg_device cpu`, view agg is run on the cpu. 
+                    Equivalently, if you define `--viewagg_device gpu`, view agg will be run on the gpu (no memory check will be done).
+* `--device <str>`: Device for processing ("cpu","cuda") (optional, default: cuda)
+* `--hires`: Switch on high resolution processing (no conforming to 1mm, but to smallest voxel size).
```
I dont think this flag still exists

agirodi replied:
This is resolved. See [here](https://github.com/Deep-MI/FastSurfer/pull/231)

dkuegler commented on FastSurferCNN/README.md:
```diff
@@ -10,66 +10,77 @@ The network was trained with conformed images (UCHAR, 1-0.7 mm voxels and standa
-* --clean: clean up segmentation after running it (optional)
-* --run_viewagg_on: Define where the view aggregation should be run on. 
+* `--clean`: clean up segmentation after running it (optional)
+* `--viewagg_device`: Define where the view aggregation should be run on. 
```
--viewagg_device <torch-compatible-device> -- this can be cpu, cuda, cuda:<device_num>, etc.

agirodi replied:
This is resolved. See [here](https://github.com/Deep-MI/FastSurfer/pull/231)

dkuegler commented on FastSurferCNN/README.md:
```diff
@@ -10,66 +10,77 @@ The network was trained with conformed images (UCHAR, 1-0.7 mm voxels and standa
-* --batch_size: Batch size for inference. Default=8
+                    If this fails, or you actively overwrote the check with setting `--viewagg_device cpu`, view agg is run on the cpu. 
+                    Equivalently, if you define `--viewagg_device gpu`, view agg will be run on the gpu (no memory check will be done).
+* `--device <str>`: Device for processing ("cpu","cuda") (optional, default: cuda)
```
this should probably be above the viewagg_device flag

agirodi replied:
This is resolved. See [here](https://github.com/Deep-MI/FastSurfer/pull/231)

agirodi committed:
docs update

- remove checkpoints in examples
- remove --seg_with_cc_only in run_fastsurfer.sh
- edit --vox_size docs
- remove hires flag
- edit --viewagg_device and --device description
- add Windows install guide

agirodi committed:
docs update tutorial

- update freesurfer version
- output update
- change CUDA out of memory fix

agirodi committed:
docs update tutorial

- fix marching_cube_lewiner not found
- add installation of packages in example
- change/clear output

agirodi committed:
docs update tutorial

- fix result print import fail
- change/clear output of QuickSeg Tutorial

agirodi committed:
docs update tutorial

- fix import fail
- add setup step for first example

agirodi committed:
Update SECURITY.md

LeHenschel committed:
Merge branch 'dev' into docs

LeHenschel committed:
Fixed errors and spelling mistakes

LeHenschel committed:
Added fastsurfer flag descriptions

LeHenschel committed:
Removed redundant code in Example 3 and 5

LeHenschel committed:
Fix flags, remove seg_with_cc_only

LeHenschel committed:
Absolute path reminder added

LeHenschel committed:
Added voxel size info, removed 256x256x256 requirement
